src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -126,32 +113,47 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
             return VSConstants.S_OK;
         }
 
-        private bool ShouldTurnOffFullSolutionAnalysis(long availableMemory)
+        private bool ShouldDisableBackgroundAnalysis(long availableMemory)
         {
             // conditions
-            // 1. if available memory is less than the threshold and 
-            // 2. if full solution analysis memory monitor is on (user can set it off using registry, when he does, we will never show info bar) and
-            // 3. if our full solution analysis option is on (not user full solution analysis option, but our internal one) and
-            // 4. if infobar is never shown to users for this solution
+            // 1. Available memory is less than the threshold and 
+            // 2. Background analysis is not already minimal and
+            // 3. Background analysis memory monitor is on (user can set it off using registry to prevent turning off background analysis)
+
             return availableMemory < MemoryThreshold &&
-                  _workspace.Options.GetOption(InternalFeatureOnOffOptions.FullSolutionAnalysisMemoryMonitor) &&
-                  _workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysis) &&
-                  !_workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysisInfoBarShown);
+                SolutionCrawlerOptions.GetBackgroundAnalysisScope(_workspace.Options) != BackgroundAnalysisScope.Minimal &&
+                _workspace.Options.GetOption(InternalFeatureOnOffOptions.BackgroundAnalysisMemoryMonitor);
+        }
+
+        private void DisableBackgroundAnalysis()
+        {
+            // Force low VM minimal background analysis for the current VS session.
+            SolutionCrawlerOptions.LowMemoryForcedMinimalBackgroundAnalysis = true;
```

Comment:
why have two options.  why not have this just set us to 'active file only' mode?  then we only have one option.

Reply:
The BG analysis scope option is a user level option that persists across VS sessions. This flag is just a temporary one for the current VS session going into minimal analysis mode when low VM is encountered, which is not persisted across sessions.